### PR TITLE
Add Boat Simulator and Enhance Build Tasks

### PR DESCRIPTION
**Pull request by DFriend01**
### Description

Related to [this issue](https://github.com/UBCSailbot/boat_simulator/issues/2). Adds the boat simulator and modifies the build script.

The following was modified:
- Added the boat simulator to the project repositories
- Modified the build script to allow for building specific packages rather than all of them at once
- Modified the VS Code tasks:
  - `Build All` is the old `Build` task
  - `Build` task prompts for specific packages to be built

### Verification

- Sailboat Simulator:
  - Rebuilt the container to see if the boat simulator repo is cloned
- Build Script:
  - The build script ran without any arguments operates as it originally did -- it builds all packages
  - `CTRL + SHIFT + B` brings up both `Build All` and `Build` tasks as build options

### CODE REVIEW

**Review by patrick-5546 — CHANGES_REQUESTED** [review]
Tested out the new build task: it works perfectly for one package, but not multiple packages.

I think this is because packages are separated by spaces, which in bash delimits arguments. My suggestion would just be to keep it simple: delete the packages task input and use the package.

I don't think it's necessary to be able to specify multiple packages to build, would be easier just build everything using the regular build task.

**Comment by DFriend01** [comment]
Wait to merge until this issue is resolved:

https://github.com/UBCSailbot/boat_simulator/issues/6

File: build.sh
Comment (patrick-5546) [inline build.sh:5]: Should be singular: `package`

File: .devcontainer/config/sailbot_workspace.code-workspace
Comment (patrick-5546) [inline .devcontainer/config/sailbot_workspace.code-workspace:218]: Should be singular: `Build a specific package...`

**Review by patrick-5546 — COMMENTED** [review]
LGTM, will approve once comments resolved and CI passes

**Review by patrick-5546 — APPROVED** [review]
CI will pass once you merge in your boat_simulator PR and rerun failed jobs

**Review by patrick-5546 — COMMENTED** [review]
One more idea that I had regarding the build tasks: incorporating `quick_build.sh`. There are multiple ways that this can be done, I'll let you decide what to do.

What do you think about combining quick_build.sh and build.sh by adding an argument to build.sh? Also if we keep quick_build.sh I think we should rename it build_quick.sh so that it is next to build.sh alphabetically. The deployment README/script would have to be updated accordingly.

File: .devcontainer/config/sailbot_workspace.code-workspace
Comment (patrick-5546) [inline .devcontainer/config/sailbot_workspace.code-workspace:254]: Also I'm thinking that it might be better to have a dedicated task for building each package. This way you don't have to type the package name every time.

Comment (patrick-5546) [reply .devcontainer/config/sailbot_workspace.code-workspace]: Used `pickString` to interactively select package from a list

**Pull request merged**
